tools/stability-checker/internal/summary/sort.go

```diff
@@ -1,45 +1,19 @@
 package summary
 
-import "strings"
-
-// By defines function comparing SpecificTestStats
-type By func(r1, r2 *SpecificTestStats) bool
-
-// ResultsSorter implements sort.Interface for SpecificTestStats
-type ResultsSorter struct {
-	List []SpecificTestStats
-	by   By
-}
-
-// NewResultsSorter returns ResultsSorter
-func NewResultsSorter(list []SpecificTestStats, by By) *ResultsSorter {
-	return &ResultsSorter{
-		List: list,
-		by:   by,
-	}
-}
+// ByMostFailures provides order of SpecificTestStats where most failed tests are placed at the begging of the slice
+type ByMostFailures []SpecificTestStats
 
 // Len is part of sort.Interface.
-func (s *ResultsSorter) Len() int {
-	return len(s.List)
+func (s ByMostFailures) Len() int {
+	return len(s)
 }
 
 // Swap is part of sort.Interface.
-func (s *ResultsSorter) Swap(i, j int) {
-	s.List[i], s.List[j] = s.List[j], s.List[i]
+func (s ByMostFailures) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
 }
 
 // Less is part of sort.Interface. It is implemented by calling the "by" closure in the sorter.
```

Comment:
outdated comment: "It is implemented by calling the "by" closure in the sorter."